### ISSUE TITLE
remove `penalty_factor` contribution to reward_factor under normal conditions

### ISSUE THREAD

**Issue by djrtwo** [ethereum/casper#86]
# Issue
Reasoning about and modeling rewards/penalties is particularly harry because `penalty_factor` makes a 2x contribution to `reward_factor` when the chain is being finalized under optimal conditions because in that case `esf() == 2`

```python
self.reward_factor = adj_interest_base + self.penalty_factor * self.esf()
```

# Suggested Implementation
change the above code to the following to remove `penalty_factor` when esf is 2

```python
self.reward_factor = adj_interest_base + self.penalty_factor * (self.esf() - 2)
```

**Comment by beneficial02** [comment]
It seems `reward_factor` can be <0 when `esf` is <2 if `penalty_factor` is >0, as `adj_interest_base` would be quite small value.

**Comment by djrtwo** [comment]
`esf` cannot be less than 2. The minimum value of esf is 2 because of how
epochs are finalized by first justifying and then finalizing.

I should have put that in the explanation of the issue

**Comment by djrtwo** [comment]
Let me further clarify that `esf` cannot be less than 2 at the time of calculating `reward_factor`. It can be equal to one at the point at which a new checkpoint is finalized in the middle of an epoch.

**Comment by djrtwo** [comment]
addressed in #87
